app/script/event/EventRepository.js

```diff
@@ -698,27 +698,74 @@ z.event.EventRepository = class EventRepository {
     return this.conversationService.update_event(identifiedUpdates);
   }
 
+  _handleDuplicatedEvent(originalEvent, newEvent) {
+    switch (newEvent.type) {
+      case z.event.Client.CONVERSATION.ASSET_ADD:
+        return this._handleAssetUpdate(originalEvent, newEvent);
+
+      case z.event.Client.CONVERSATION.MESSAGE_ADD:
+        return this._handleLinkPreviewUpdate(originalEvent, newEvent);
+
+      default:
+        this._throwValidationError(newEvent, `Forbidden type '${newEvent.type}' for duplicate events`);
+    }
+  }
+
+  _handleAssetUpdate(originalEvent, newEvent) {
+    const newEventData = newEvent.data;
+    // the preview status is not sent by the client so we fake a 'preview' status in order to cleany handle it in the switch statement
+    const LOCAL_PREVIEW_STATUS = 'preview';
```

Comment:
Let's come up with a better name. This status is not local. Only the variable is. It used to be a generic message proto of type asset with content preview. So maybe we should just call it `ASSET_PREVIEW`. Wdyt?